UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs

```diff
@@ -54,5 +54,20 @@ public async Task<List<string>> GetYears()
                                 .Select(p => p.DateTimeGroup.Year.ToString())
                                 .Distinct().ToListAsync();
         }
+
+        public async Task<List<RadioNavigationalWarningsData>> GetRadioNavigationalWarningsDataList()
+        {
+            return await (from rnwWarnings in _context.RadioNavigationalWarnings
+                     join warning in _context.WarningType on rnwWarnings.WarningType equals warning.Id
+                     where !rnwWarnings.IsDeleted && rnwWarnings.ExpiryDate >= DateTime.UtcNow
```

Comment:
Records with ExpiryDate is null will never be returned. Please update where clause to return records with ExpiryDate is null.
```suggestion
                     where !rnwWarnings.IsDeleted && (rnwWarnings.ExpiryDate == null || rnwWarnings.ExpiryDate >= DateTime.UtcNow)
```

Reply:
Done required changes